### DISCUSSION TITLE
BROKEN status

### DISCUSSION THREAD

**Discussion by sergeidyga** [ota4j-team/open-test-reporting#49] [General]
Does it make sense to support BROKEN status in schema. With BROKEN meaning any failure != AssertionError.

**Comment by peturingi** [comment]
I strongly advise against introducing a BROKEN status into the schema.

The project’s stated goal is _"XML and HTML test reporting formats that are agnostic of testing framework and programming language."_ Introducing BROKEN would violate this principle by imposing a classification tied to exception semantics, which vary significantly across testing-frameworks/languages.

To take [pytest](https://docs.pytest.org/en/stable/) (a popular [python](https://www.python.org) testing framework) as an example; it specifically distinguishes between three outcomes:
• PASSED - test, including setup and teardown logic, executed as expected,
• FAILED - any unhandled exception in the test body, whether it’s an AssertionError, TypeError, ValueError, or any other Exception.
• ERROR - exceptions raised by non-test-body logic.

Notably, pytest does not differentiate between assertion failures and other exception types within the test body -- all are uniformly reported as FAILED or ERROR depending on whether it was the test itself that FAILED, or ERROR occurred in relation to the test execution.

This PASSED / FAILED / ERROR triplet is a widely adopted de facto standard that maps directly to multiple mainstream testing frameworks, including:

pytest: PASSED / FAILED / ERROR (fixtures, collection)
JUnit: PASSED / FAILED / ERROR (@Before/@After exceptions)
TestNG: PASSED / FAILED / ERROR (config method failures)
NUnit: PASSED / FAILED / ERROR (setup/teardown issues)
GoogleTest: PASSED / FAILED / ERROR (fatal env/setup failures)
Robot Framework: PASSED / FAILED / ERROR (setup/teardown/keyword failures)

CI systems such as Jenkins, GitLab CI, CircleCI, and Azure DevOps also interpret test results using this established triplet when generating summaries and test reports.

Adding a BROKEN status defined as "non-assertion exception" introduces _semantic ambiguity_ because:
• most frameworks do not distinguish exceptions in this manner.,
• many languages lack a standardised assertion exception class, and
• it would necessitate post hoc exception introspection which arguably compromises the project's neutrality.

The current (version 0.2.0) definition of Status maps conveniently to this widely adopted PASSED / FAILED / ERROR triplet, maintaining neutrality and interoperability across testing-frameworks/languages.
—
**Pétur**
